### DISCUSSION TITLE
Coolify local deployment tool?

### DISCUSSION THREAD

**Discussion by bchubb-web** [coollabsio/coolify#3072] [Q&A]
I was wondering if there is any plan to use coolify as a local environment manager, im not sure about other peoples workflows but i would find it very useful to have the same tool managing all aspects of deploying an application.

I'm not sure what approach would be best but my intial idea would be a locally deployable coolify instance, which provides a gui and cli to deploy any application in the same way it would to a remote server currently.

**Comment by peaklabs-dev** [comment]
@bchubb-web You could deploy it locally on a Linux VM for example and use it the same way as in the cloud. Only DNS and HTTPS are a bit trickier, but just use Cloudflare tunnels and it should work.

**Reply by hayer** [reply]
@peaklabs-dev and is there anyway to do this without Cloudflare tunnels?
For when you only want to expose the services hosted to an internal network.

**Reply by Cinzya** [reply]
@hayer You can in theory just point your domain to your internal ip address. As long as port 80 and 443 are accessible through the internal network.
